src/einsteinpy/coordinates/core.py

```diff
@@ -30,13 +30,36 @@ def __init__(self, x, y, z):
         self.z = z
         super().__init__(x.si.value, y.si.value, z.si.value)
         self.system = "Cartesian"
+        self._dimension = {
+            "x": self.x,
+            "y": self.y,
+            "z": self.z,
+            "system": "Cartesian",
+        }
+        self._dimension_order = ("x", "y", "z")
 
     def __repr__(self):
         return "Cartesian x: {}, y: {}, z: {}".format(self.x, self.y, self.z)
 
     def __str__(self):
         return self.__repr__()
 
+    def __getitem__(self, item):
+        """
+        Method to return coordinates. 
+        Objects are subsctiptable with both explicit names of parameters
+        and integer indices.
+
+        Parameters
+        ----------
+        item : ~astropy.units.quantity.Quantity or integer
+                Name of the parameter or it's index.
+                If `system`, Name of coordinate is returned.
+        """
+        if isinstance(item, (int, np.integer)):
```

Comment:
Very good @Raahul-Singh ! This is exactlt what was expected!

Reply:
Thank you for helping me along the way!